src/Symfony/Component/CssSelector/Tests/XPath/TranslatorTest.php

```diff
@@ -219,6 +219,7 @@ public static function getCssToXPathTestData()
             ['e + f', "e/following-sibling::*[(name() = 'f') and (position() = 1)]"],
             ['e ~ f', 'e/following-sibling::f'],
             ['div#container p', "div[@id = 'container']/descendant-or-self::*/p"],
+            ['div:has(div.foo)', "div[count(descendant-or-self::*[(@class and contains(concat(' ', normalize-space(@class), ' '), ' foo ')) and (name() = 'div')]) > 0]"],
```

Comment:
we should also have tests covering more complex selectors in `:has()`, as well as relative selectors: `div:has(.foo .bar)`, `div:has(> .foo)`, `div:has(~ foo)`